pkg/catalogv2/git/utils.go

```diff
@@ -0,0 +1,207 @@
+package git
+
+import (
+	"crypto/sha256"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/hex"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type basicRoundTripper struct {
+	username string
+	password string
+	next     http.RoundTripper
+}
+
+func (b *basicRoundTripper) RoundTrip(request *http.Request) (*http.Response, error) {
+	request.SetBasicAuth(b.username, b.password)
+	return b.next.RoundTrip(request)
+}
+
+func gitDir(namespace, name, gitURL string) string {
+	staticDir := filepath.Join(staticDir, namespace, name, hash(gitURL))
+	if s, err := os.Stat(staticDir); err == nil && s.IsDir() {
+		return staticDir
+	}
+	return filepath.Join(stateDir, namespace, name, hash(gitURL))
```

Comment:
The URL was being hashed twice:
```suggestion
        h := hash(gitURL)
	staticDir := filepath.Join(staticDir, namespace, name, h)
	if s, err := os.Stat(staticDir); err == nil && s.IsDir() {
		return staticDir
	}
	return filepath.Join(stateDir, namespace, name, h)
```